From a778d5b035e3c147d0938190559e35ba4b278ed5 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Mon, 13 Oct 2014 15:51:34 +0200
Subject: [PATCH 067/117] ls: in CSV mode, always have a checksum field
 (RHBZ#1151900).

Make sure to output the field for checksum even for non-regular files,
as empty, in CSV output mode. This ensures each line has the same number
of fields, regardless of the file type.

(cherry picked from commit e5befcacc5f7c1a6bd67fcdb92afede1a82d50eb)
---
 cat/virt-ls.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/cat/virt-ls.c b/cat/virt-ls.c
index 4eac5c8..e6dea23 100644
--- a/cat/virt-ls.c
+++ b/cat/virt-ls.c
@@ -626,12 +626,15 @@ show_file (const char *dir, const char *name,
 
   path = full_path (dir, name);
 
-  if (checksum && is_reg (stat->mode)) {
-    csum = guestfs_checksum (g, checksum, path);
-    if (!csum)
-      exit (EXIT_FAILURE);
+  if (checksum) {
+    if (is_reg (stat->mode)) {
+      csum = guestfs_checksum (g, checksum, path);
+      if (!csum)
+        exit (EXIT_FAILURE);
 
-    output_string (csum);
+      output_string (csum);
+    } else if (csv)
+      output_string ("");
   }
 
   output_string (path);
-- 
2.5.0

